src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EcDsa.ImportExport.cs

```diff
@@ -84,8 +84,9 @@ internal static SafeEcKeyHandle EcKeyCreateByExplicitCurve(ECCurve curve)
 
             if (key == null || key.IsInvalid)
             {
+                Exception e = Interop.Crypto.CreateOpenSslCryptographicException();
```

Comment:
This pattern is reoccurring, is disposing of the handling impacting the computed exception?

Reply:
> This pattern is reoccurring, is disposing of the handling impacting the computed exception?

In some cases I believe it could.  OpenSSL, for example, works on a stack of error codes, and any operation (including resource release) could in theory push an error on to the stack; then when we retrieve the last one to throw the exception, we'd get the wrong result.

In general, executing arbitrary code between when an error is ambiently stored and subsequently retrieved seems like a recipe for sometimes getting it wrong.